### ISSUE TITLE
Better DEV warnings for invalid props

### ISSUE THREAD

**Issue by bvaughn** [bvaughn/react-resizable-panels#65]
### Invalid default sizes
The following panel group has an invalid configuration, which will cause _flickering_ during resize:
```jsx
<PanelGroup direction="horizontal">
  <Panel defaultSize={20}>
    {/* ... */}
   </Panel>
  <ResizeHandle />
  <Panel defaultSize={20}>
    {/* ... */}
   </Panel>
</PanelGroup>
```

### Panel with `onResize` prop and no `order` prop
The docs warn about this but we could also programmatically detect and warn as well since it is non-obvious.

**Comment by bvaughn** [comment]
Looks like it's because the Sandbox I forked had a group with two panels, both of which had `defaultSize={20}` – should add a warning for this cause.

**Comment by ethanyou725** [comment]
inappropriate props `maxSize` and `minSize` will cause unexpected behavior when call  `onCollapse`, it may need a dev warning too.

**Comment by bvaughn** [comment]
I don't know how to accomplish this with Parcel at the moment.

Stashing an attempt here for later:
```diff
diff --git a/packages/react-resizable-panels/package.json b/packages/react-resizable-panels/package.json
index f05b77a..689be07 100644
--- a/packages/react-resizable-panels/package.json
+++ b/packages/react-resizable-panels/package.json
@@ -28,6 +28,12 @@
     "react": "^16.14.0 || ^17.0.0 || ^18.0.0",
     "react-dom": "^16.14.0 || ^17.0.0 || ^18.0.0"
   },
+  "engines": {
+    "node": ">=12"
+  },
+  "@parcel/transformer-js": {
+    "inlineEnvironment": false
+  },
   "browserslist": [
     "Chrome 79"
   ]
diff --git a/packages/react-resizable-panels/src/PanelGroup.ts b/packages/react-resizable-panels/src/PanelGroup.ts
index d860581..dd4273f 100644
--- a/packages/react-resizable-panels/src/PanelGroup.ts
+++ b/packages/react-resizable-panels/src/PanelGroup.ts
@@ -45,6 +45,7 @@ import {
   panelsMapToSortedArray,
 } from "./utils/group";
 import { loadPanelLayout, savePanelGroupLayout } from "./utils/serialization";
+import { isServerRendering } from "./utils/ssr";
 
 const debounceMap: {
   [key: string]: (
@@ -316,6 +317,14 @@ function PanelGroupWithForwardedRef({
       // This includes server rendering.
       // At this point the best we can do is render everything with the same size.
       if (panels.size === 0) {
+        if (process.env.NODE_ENV !== "production") {
+          if (isServerRendering() && defaultSize == null) {
+            console.warn(
+              `WARNING: Panel defaultSize prop recommended to avoid layout shift after server rendering`
+            );
+          }
+        }
+
         return {
           flexBasis: 0,
           flexGrow: defaultSize != null ? defaultSize : undefined,
diff --git a/packages/react-resizable-panels/src/utils/ssr.ts b/packages/react-resizable-panels/src/utils/ssr.ts
new file mode 100644
index 0000000..878fcf0
--- /dev/null
+++ b/packages/react-resizable-panels/src/utils/ssr.ts
@@ -0,0 +1,7 @@
+export function isServerRendering(): boolean {
+  try {
+    return typeof window === undefined;
+  } catch (error) {}
+
+  return true;
+}
```

**Comment by bvaughn** [comment]
Here's a good thread: https://twitter.com/AndaristRake/status/1658106033603067905

**Comment by bvaughn** [comment]
Warnings are now in DEV bundles only 🥳

**Comment by bvaughn** [comment]
Resolved as of 3780fd3
